diff --git a/data/data/libvirt/main.tf b/data/data/libvirt/main.tf
index 680919974..37307374f 100644
--- a/data/data/libvirt/main.tf
+++ b/data/data/libvirt/main.tf
@@ -62,6 +62,7 @@ resource "libvirt_network" "net" {
         data.libvirt_network_dns_host_template.masters.*.rendered,
         data.libvirt_network_dns_host_template.masters_int.*.rendered,
         data.libvirt_network_dns_host_template.etcds.*.rendered,
+	data.libvirt_network_dns_host_template.auth.*.rendered,
       )
       content {
         hostname = hosts.value.hostname
@@ -133,6 +134,12 @@ data "libvirt_network_dns_host_template" "etcds" {
   hostname = "etcd-${count.index}.${var.cluster_domain}"
 }
 
+data "libvirt_network_dns_host_template" "auth" {
+  count    = var.master_count
+  ip       = var.libvirt_auth_ip
+  hostname = "oauth-openshift.apps.${var.cluster_domain}"
+}
+
 data "libvirt_network_dns_srv_template" "etcd_cluster" {
   count    = var.master_count
   service  = "etcd-server-ssl"
diff --git a/data/data/libvirt/variables-libvirt.tf b/data/data/libvirt/variables-libvirt.tf
index 53cf68bae..c39df9f4d 100644
--- a/data/data/libvirt/variables-libvirt.tf
+++ b/data/data/libvirt/variables-libvirt.tf
@@ -35,6 +35,12 @@ variable "libvirt_master_memory" {
   default     = "6144"
 }
 
+variable "libvirt_auth_ip" {
+  type        = "string"
+  description = "node with authentication server ip"
+  default     = "192.168.126.51"
+}
+
 # At some point this one is likely to default to the number
 # of physical cores you have.  See also
 # https://pagure.io/standard-test-roles/pull-request/223
diff --git a/pkg/tfvars/libvirt/libvirt.go b/pkg/tfvars/libvirt/libvirt.go
index 4b7b7f50b..056bc2de1 100644
--- a/pkg/tfvars/libvirt/libvirt.go
+++ b/pkg/tfvars/libvirt/libvirt.go
@@ -17,6 +17,7 @@ type config struct {
 	IfName      string   `json:"libvirt_network_if"`
 	MasterIPs   []string `json:"libvirt_master_ips,omitempty"`
 	BootstrapIP string   `json:"libvirt_bootstrap_ip,omitempty"`
+	AuthNodeIP  string   `json:"libvirt_auth_ip,omitempty"`
 }
 
 // TFVars generates libvirt-specific Terraform variables.
@@ -42,6 +43,7 @@ func TFVars(masterConfig *v1beta1.LibvirtMachineProviderConfig, osImage string,
 		IfName:      bridge,
 		BootstrapIP: bootstrapIP.String(),
 		MasterIPs:   masterIPs,
+		AuthNodeIP:  "192.168.126.51",
 	}
 
 	return json.MarshalIndent(cfg, "", "  ")
